devito/types/dense.py

```diff
@@ -734,7 +734,6 @@ def _arg_names(self):
         """Tuple of argument names introduced by this function."""
         return (self.name,)
 
-    @memoized_meth
```

Comment:
why?

Reply:
With MPI enabled it results in halo updates not always occurring when they need to. But some other stuff has been memoized now though so that the performance when calling an operator repeatedly it not affected too much.

Reply:
Do we have a test for this? Im sure in one year time I’ll try to add it back because Ill have forgotten about this MPI detail. Also a comment “Note: this, unlike all other _arg_defaults, cant be memoized because ...” would help

Reply:
Good idea. Will add the test + note.